### PR TITLE
Readjust reputation gain on lower level mobs/quests

### PR DESCRIPTION
**Pull request by Phatcat**
The reputation system rate on lower level quests and mobs is more complex than simply having 1 config value for gray levels, which doesn't make much sense.

> Just like XP, if the mob or quest level is 6 or more levels below you, the reputation gained will be lowered by 20% for each level (however it can't be lower than 20%). Completing quests or killing mobs whose level is higher than yours won't give you more reputation.
> 
> For example, completing a level 10 quest that normally gives 25 points will give: 
> 
> Level | Points
> 1 | 25 (100%)
> 10 | 25 (100%)
> 15 | 25 (100%)
> 16 | 20 (80%)
> 17 | 15 (60%)
> 18 | 10 (40%)
> 19 | 5 (20%)
> 60 | 5 (20%)

https://wowwiki.fandom.com/wiki/Reputation?oldid=346091#Gaining_and_Losing_reputation

This was changed in wotlk to not have any penalty to reputation gains from lower level mobs/quests;

>Full reputation is still gained by questing at a higher level. For example, completing low level quests with a level 80 character now gives full reputation. This makes it easier to get to exalted and buy another faction's mount.
>
>Grey mobs used to give 20% of the normal reputation until the release of Patch 3.0.8. Now any mob will give full reputation. See the mob difficulty colors chart for when mobs transition from green to grey.

https://wowwiki.fandom.com/wiki/Reputation#Gaining_and_losing_reputation

### CODE REVIEW

File: src/mangosd/mangosd.conf.dist.in [hunk 1]
```diff
@@ -1279,12 +1279,12 @@ Visibility.AIRelocationNotifyDelay = 1000
...
 #    Rate.Reputation.LowLevel.Kill
-#         Reputation Gain form low level kill (grey creture)
+#         Lowest Reputation Gain rate from low level kill
 #         Default: 0.2
 #
 #    Rate.Reputation.LowLevel.Quest
-#         Reputation Gain rate
-#         Default: 1
+#         Lowest Reputation Gain rate from low level quest
+#         Default: 0.2
```
[the remark this answers is not in the document]
Comment (Phatcat) [reply src/mangosd/mangosd.conf.dist.in]: Desired new default should be 0.2 for both quests and kills for classic, not 1 and 0.2
This is why I changed it in the config file to be so. I did forget about the load during world honestly, yeah, I'll have that fixed.

File: src/game/Entities/Player.cpp [hunk 2]
```diff
@@ -5882,22 +5882,46 @@ int32 Player::CalculateReputationGain(ReputationSource source, int32 rep, int32
...
             break;
+        }
         case REPUTATION_SOURCE_SPELL:
         default:
             rate = 1.0f;
             break;
     }
 
-    if (rate != 1.0f && creatureOrQuestLevel <= MaNGOS::XP::GetGrayLevel(getLevel()))
+    if (rate != 1.0f)
```
[the remark this answers is not in the document]
Comment (Phatcat) [reply src/game/Entities/Player.cpp]: What do you mean it is equalized, though? It's a float on float comparison, rate is a float not a uin32 and 1.0f is a float, where is the issue happening here? is it the != operation? Hmm..

[the remark this answers is not in the document]
Comment (Phatcat) [reply src/game/Entities/Player.cpp]: It should work as long as I am not doing a comparison against 1 or 1.0 but do it specifically against 1.0f like I am doing.

https://www.geeksforgeeks.org/comparison-float-value-c/

[the remark this answers is not in the document]
Comment (norcimoix) [reply src/game/Entities/Player.cpp]: Depends where the numbers come from I suppose. Generally you don't use equality or inequality comparisons on floating point numbers because rounding errors can make them be slightly off:

    >>> 0.3 - 0.2 == 0.1
    False

(This is from the python interpreter but they use the same IEEE754 floating point standard that intel CPUs and pretty much the whole world use)

However, if no operations are done on the floats then the comparisons will never cause issues, especially not on floats like 1.0 that have an exact representation.

TL;DR: I don't think this particular case will cause any issues.

**Comment by Phatcat** [comment]
The whole thing with the treshold is a lot nicer than the if/else I was doing, I basically just copied it over from where the quest xp calculations take place, which probably also could do with a touch-up since it does some weird undocumented calculations on that..

Anyways I've updated the PR according to your changes (with a small adjustment)
